runtime/parachains/src/configuration.rs

```diff
@@ -1054,9 +1065,28 @@ impl<T: Config> Pallet<T> {
 
 	/// Called by the initializer to note that a new session has started.
 	pub(crate) fn initializer_on_new_session(session_index: &SessionIndex) {
-		if let Some(pending) = <Self as Store>::PendingConfig::take(session_index) {
-			<Self as Store>::ActiveConfig::set(pending);
+		let pending_configs = <PendingConfigs<T>>::get();
+		if pending_configs.is_empty() {
+			return
 		}
+
+		let (past_and_now, future) = pending_configs
```

Comment:
```suggestion
		let (past_and_present, future) = pending_configs
```

Extreme nit pick, but this the associated word with "past" and "future" it "present"